plugins/ccgx/fu-ccgx-dmc-device.c

```diff
@@ -616,33 +616,44 @@ fu_ccgx_dmc_device_attach(FuDevice *device, FuProgress *progress, GError **error
 	if (fu_device_get_update_state(self) != FWUPD_UPDATE_STATE_SUCCESS)
 		return TRUE;
 
+	if (manual_replug) {
+		fu_device_add_flag(device, FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION);
+		fu_device_inhibit(device,
+				  "update-pending",
+				  "A pending update will be completed next time the device "
+				  "is unplugged from your computer");
+	} else {
+		fu_device_add_flag(device, FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);
+	}
+
 	if (self->update_model == DMC_UPDATE_MODEL_DOWNLOAD_TRIGGER) {
 		if (self->trigger_code > 0) {
 			if (!fu_ccgx_dmc_device_send_download_trigger(self,
 								      self->trigger_code,
 								      error)) {
+				if (manual_replug) {
+					fu_device_remove_flag(device,
+							      FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION);
+				} else {
+					fu_device_remove_flag(device,
+							      FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);
```

Comment:
I understand removing FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG -- but do you really want to remove FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION too? It seems the hardware still needs activating, even if the reset failed, no? If you also want to remove FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION you also need to uninhibit  `update-pending` too.

Reply:
FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION is set when device need manual re-plug such as Lenovo USB-C dock.
Is it OK not to remove "FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION" flag when return FALSE in attach?
If yes, I will not remove "FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION" flag.

Reply:
@hughsie
FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION flag remain.
See [932298a](https://github.com/fwupd/fwupd/pull/4563/commits/932298a793b135c88c2db6bc866e35f2107f57cb)